src/test/java/org/prebid/server/bidder/kobler/KoblerBidderTest.java

```diff
@@ -0,0 +1,284 @@
+            assertThat(error.getType()).isEqualTo(BidderError.Type.bad_server_response);
+            assertThat(error.getMessage()).startsWith("Failed to decode");
+        });
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyListWhenBidResponseIsNull() {
+        // given
+        final BidderCall<BidRequest> httpCall = givenHttpCallWithBody("null");
+
+        // when
+        final Result<List<BidderBid>> result = target.makeBids(httpCall, null);
+        // then
+        assertThat(result.getValue()).isEmpty();
+        assertThat(result.getErrors()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyListWhenSeatbidIsEmpty() throws JsonProcessingException {
+        // given
+        final BidderCall<BidRequest> httpCall = givenHttpCall(givenBidResponse());
+
+        // when
+        final Result<List<BidderBid>> result = target.makeBids(httpCall, null);
+        // then
+        assertThat(result.getValue()).isEmpty();
+        assertThat(result.getErrors()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnBidWithBannerType() throws JsonProcessingException {
+        // given
+        final Bid bid = Bid.builder()
+                .ext(mapper.valueToTree(Map.of("prebid", Map.of("type", "banner"))))
+                .build();
+
+        final BidderCall<BidRequest> httpCall = givenHttpCall(givenBidResponse(bid));
+
+        // when
+        final Result<List<BidderBid>> result = target.makeBids(httpCall, BidRequest.builder().build());
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
+                .extracting(BidderBid::getType)
+                .containsExactly(BidType.banner);
+    }
+
+    @Test
+    public void makeBidsShouldReturnBannerWhenTypeIsNullOrMissing() throws JsonProcessingException {
+        // given
+        final Bid bid = Bid.builder()
+                .ext(mapper.valueToTree(Map.of("prebid", Map.of("type", "null"))))
+                .build();
+
+        final BidderCall<BidRequest> httpCall = givenHttpCall(givenBidResponse(bid));
+
+        // when
+        final Result<List<BidderBid>> result = target.makeBids(httpCall, BidRequest.builder().build());
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
+                .extracting(BidderBid::getType)
+                .containsExactly(BidType.banner);
+    }
+
+    @Test
+    public void makeBidsShouldDefaultToBannerWhenPrebidTypeIsMissing() throws JsonProcessingException {
+        // given
+        final Bid bid = Bid.builder()
+                .ext(mapper.valueToTree(Map.of("prebid", Map.of())))
+                .build();
+
+        final BidderCall<BidRequest> httpCall = givenHttpCall(givenBidResponse(bid));
+
+        // when
+        final Result<List<BidderBid>> result = target.makeBids(httpCall, BidRequest.builder().build());
```

Comment:
Please remove unnecessary `BidRequest` objects. I have written comments about this in other places, but accidentally missed it in these tests.